### PR TITLE
add warmup and horovod support (Inception V3 could reach to 77.6% …

### PR DESCRIPTION
**Pull request by fzou1**
…test accuracy with 2K global batch size, 32 per CPU rank); add number of intra and inter threads to get better training performance on CPU

### CODE REVIEW

**Comment by fzou1** [comment]
@sguada May I ask your helps for review? Thanks.

**Comment by ejjeong** [comment]
@fzou1 I’m interested in training inception v3 network on distributed environment, and it would really appreciated if you can share your command-line arguments to achieve the 77.6% accuracy with this PR.
(Especially I want to know the arguments related to the hyperparameters, including `learning_rate`, `moving_average_decay` for EMA, `label_smoothing`, `warmup_epochs`, … etc.)

**Comment by fzou1** [comment]
@ejjeong May I know what's your usage? We only ensure this works on Intel CPU.

**Comment by ejjeong** [comment]
I'm just trying to find the right hyperparameters to train the inception v3 model when using larger batch sizes. (The ones in CVPR inception v3 paperdidn't seem to work, at least for the model in slim directory.) I use the NVidia GPUs, but I think your command line arguments would still be very helpful on the same batch size :)

**Comment by wei-v-wang** [comment]
Hi @sguada, could you please help review this PR? Thank you! 
This improves inception-v3, inception-v4, and inception-resnetv2 at least. Thank you!
